MCFM-JHUGen/src/ZZ/getggZZamps.f

```diff
@@ -157,13 +159,41 @@ subroutine getggZZamps(p,dolight,dobottom,dotop,
       cl1(2)=r1
       cl2(1)=l2
       cl2(2)=r2
-
+c --- Modifications to lepton couplings as arrays for anomalous Zll couplings 
+c --- Modified by Jeff
+      if (AllowAnomalousZffCouplings .eq. 1) then
+        if ((plabel(3) .eq. 'el') .or. (plabel(3) .eq. 'ml')
+     &.or. (plabel(3) .eq. 'tl')) then
+          cl1(1) = leZ
+          cl1(2) = reZ 
+        endif
+        elseif (plabel(3) .eq. 'nl') then
+          cl1(1) = lnZ*dsqrt(3d0)
+          cl1(2) = rnZ*dsqrt(3d0) 
+        endif
+      if (AllowAnomalousZffCouplings .eq. 1) then
+        if ((plabel(5) .eq. 'el') .or. (plabel(5) .eq. 'ml')
+     &.or. (plabel(5) .eq. 'tl')) then
+          cl2(1) = leZ
+          cl2(2) = reZ 
+        endif
+        elseif (plabel(5) .eq. 'nl') then
+          cl2(1) = lnZ*dsqrt(3d0)
+          cl2(2) = rnZ*dsqrt(3d0) 
+        endif  
```

Comment:
Insufficient set of if-conditions. Matrix elements also handle quark final states.